### PR TITLE
wluma: 4.0.0 -> 4.1.0

### PR DESCRIPTION
**Pull request by jmc-figueira**
###### Motivation for this change

Updated the package version from 4.0.0 to 4.1.0.

This version added documentation (in the form of manpages) to `wluma`. However, [building this documentation](https://github.com/maximbaz/wluma/blob/a94209094256df23b581a01bc9295370e9fbaebe/Makefile#L36) depends on `marked-man`, an utility which, as far as I could find, is not packaged in nixpkgs. If this is a requirement for packaging, I may attempt to package `marked-man` as well; however, having no experience with packaging node.js packages, it may take a bit more time.

I also added myself as maintainer for this package, as it seems like a rather simple package to maintain as a new maintainer, coupled with the fact that it is written in Rust, a language I am familiar with.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jmc-figueira** [comment]
Result of `nixpkgs-review pr 156112` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wluma</li>
  </ul>
</details>

**Comment by yevhenshymotiuk** [comment]
Yes, `marked-man` should be packaged. Even though it just uses `README.md` as an input, I don't think we should skip it

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5709,6 +5709,26 @@
...
+    githubId = 6634716;
+    name = "João Figueira";
+    keys = [
+
+      # GitHub signing key
+      {
+        longkeyid = "rsa4096/0xDC7AE56AE98E02D7";
+        fingerprint = "EC08 7AA3 DEAD A972 F015  6371 DC7A E56A E98E 02D7";
+      }
+
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5724]:
```suggestion
    keys = [
      # GitHub signing key
      {
        longkeyid = "rsa4096/0xDC7AE56AE98E02D7";
        fingerprint = "EC08 7AA3 DEAD A972 F015  6371 DC7A E56A E98E 02D7";
      }
```

**Comment by yevhenshymotiuk** [comment]
@SuperSandro2000 Why was this merged?

**Comment by jmc-figueira** [comment]
Just a status update: I am still working on packaging `marked-man` (having encountered some issues along the way). Once it's done, I was thinking of opening a new PR to correct this issue, would that be the best way of handling this situation?

**Comment by yevhenshymotiuk** [comment]
> Just a status update: I am still working on packaging `marked-man` (having encountered some issues along the way). Once it's done, I was thinking of opening a new PR to correct this issue, would that be the best way of handling this situation?

Sounds good

**Comment by maximbaz** [comment]
Guys FYI in 4.1.2 I followed what Alpine's packager has done and started linking against system Vulkan and Wayland. 

This means that instead of using Rust implementations, the app will use "original" implementations from system libraries (less bugs, hopefully?).

Both Vulkan and Wayland are technically optional for wluma (app should work fine if you just use `capturer=none` in config), so here's how that is handled:

- For Vulkan you add a dependency for ICD loader, but it's up to user to actually install a specific Vulkan driver if they want to use `capturer=wlroots`
- For Wayland the code should be able to dynamically load the library only if `capturer=wlroots` is used, so again it's up to user to install the dependency if they need that functionality.

For Arch pkg I simply made these dependencies [optional](https://github.com/maximbaz/pkgbuilds/commit/176aea8165a8a6877e9a3748665f51aaff4a4a87#diff-6d88240e8b20e8fadd7de07616213ca88cafe36dccc12c4548babee773eab36e) (they weren't really required before, as Rust impl was used).

See what's best for Nix, and do let me know if you discover any issues, I'll do my best to help out 😉

**Comment by SuperSandro2000** [comment]
> See what's best for Nix, and do let me know if you discover any issues, I'll do my best to help out 😉

Gate it behind an boolean input or just add it all the time regardless of the config.

**Comment by yshym** [comment]
> Gate it behind an boolean input or just add it all the time regardless of the config.

Yes, I think we should add a boolean function parameter to allow users to choose whether to include these dependencies or not

**Pull request merged**